repair/row_level.cc

```diff
@@ -548,21 +553,31 @@ class repair_writer {
         }
     }
 
+    future<> write_end_of_stream(unsigned node_idx) {
+        // Partition_end is never sent on wire, so we have to write one ourselves.
+        return write_partition_end(node_idx).then([this, node_idx] () mutable {
+            // Empty mutation_fragment_opt means no more data, so the writer can seal the sstables.
+            return _mq[node_idx]->push_eventually(mutation_fragment_opt());
+        });
+    }
+
+    future<> do_wait_for_writer_done(unsigned node_idx) {
+        return (*_writer_done[node_idx]).then([] (uint64_t partitions) {
+            rlogger.debug("Managed to write partitions={} to sstable", partitions);
+            return make_ready_future<>();
+        });
```

Comment:
Why no do success logging in the same continuation as a failure. That you will not even need this function, simply move _writer_done directly to when_all().

Reply:
I did not get you. What do you mean? Are you complaining we should not have the debug log here?